### PR TITLE
logging: insert request into notice params

### LINKED ISSUE #94: Django's request object is not inserted into the context on log.exception
When Django is available pybrake should save request in thread local storage as described in https://nedbatchelder.com/blog/201008/global_django_requests.html and use it in then notice filter to report request data as in https://github.com/airbrake/pybrake/blob/master/pybrake/django.py#L86-L95

### PR DESCRIPTION
**Pull request by anmic**
Fixes #94.

### CODE REVIEW

**Comment by anmic** [comment]
Added request filter.

File: pybrake/logging.py [hunk 1]
```diff
@@ -32,6 +33,10 @@ def __init__(self, notifier=None, level=logging.ERROR, **kwargs):
         logging.Handler.__init__(self, level=level)
         if notifier is None:
             notifier = get_global_notifier() or Notifier(**kwargs)
+
+        if _DJANGO:
+            notifier.add_filter(request_filter)
```
Comment (vmihailenco) [inline pybrake/logging.py]: Why this is added here?

File: pybrake/django.py [hunk 2]
```diff
@@ -16,6 +17,31 @@
 
 _UNKNOWN_ROUTE = "UNKNOWN"
 
+requests = threading.local()
```
Comment (vmihailenco) [inline pybrake/django.py]: rename requests

File: pybrake/django.py [hunk 3]
```diff
@@ -16,6 +17,31 @@
...
+
+def set_request(request):
+    setattr(requests, "request", request)
+
+def get_request():
+    return getattr(requests, "request", None)
+
+def request_filter(notice):
+    request = get_request()
```
Comment (vmihailenco) [inline pybrake/django.py]: what if request is none?

File: pybrake/django.py [hunk 4]
```diff
@@ -36,6 +62,8 @@ def __init__(self, get_response):
         self._apm_disabled = self._notifier._apm_disabled
 
     def __call__(self, request):
+        set_request(request)
```
Comment (vmihailenco) [inline pybrake/django.py]: set request to None after it is served

**Comment by anmic** [comment]
Moved Django request filter to separate module.

File: pybrake/django_filter.py [hunk 5]
```diff
@@ -0,0 +1,45 @@
+import threading
```
Comment (vmihailenco) [inline pybrake/django_filter.py]: move this code to django.py

File: pybrake/django_filter.py [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+def set_request(request):
+    setattr(threadLocal, _REQUEST_KEY, request)
+
+def get_request():
+    return getattr(threadLocal, _REQUEST_KEY, None)
+
+def request_filter(notice):
+    request = getattr(threadLocal, _REQUEST_KEY, None)
+    setattr(threadLocal, _REQUEST_KEY, None)
```
Comment (vmihailenco) [inline pybrake/django_filter.py]: this does not make much sense

File: pybrake/django.py [hunk 7]
```diff
@@ -112,18 +142,6 @@ def get_remote_addr(request):
...
-
-@functools.lru_cache()
-def get_default_exception_reporter_filter():
-    # Instantiate the default filter for the first time and cache it.
-    return import_string(settings.DEFAULT_EXCEPTION_REPORTER_FILTER)()
-
-
-def get_exception_reporter_filter(request):
-    default_filter = get_default_exception_reporter_filter()
-    return getattr(request, "exception_reporter_filter", default_filter)
```
Comment (vmihailenco) [inline pybrake/django.py]: revert this change

File: pybrake/global_notifier.py [hunk 8]
```diff
@@ -22,5 +22,8 @@ def get_global_notifier():
             return None
 
     from .notifier import Notifier
+    notifier = Notifier(**settings.AIRBRAKE)
 
-    return Notifier(**settings.AIRBRAKE)
+    from .django import request_filter
+    notifier.add_filter(request_filter)
```
Comment (vmihailenco) [inline pybrake/global_notifier.py]: why this is added here?

File: pybrake/django.py [hunk 9]
```diff
@@ -68,6 +108,8 @@ def process_view(self, request, view_func, view_args, view_kwargs):
             route += "." + view_func.__name__
             metric.route = route
 
+        request = None
```
Comment (vmihailenco) [inline pybrake/django.py]: add a comment what this code does

Comment (anmic) [reply pybrake/django.py]: I tried to set the mutable object to None, but that's not how it works.

**Comment by anmic** [comment]
Fixed import cycle.

File: pybrake/logging.py [hunk 10]
```diff
@@ -34,6 +35,9 @@ def __init__(self, notifier=None, level=logging.ERROR, **kwargs):
             notifier = get_global_notifier() or Notifier(**kwargs)
         self._notifier = notifier
 
+        if _DJANGO:
+            self._notifier.add_filter(request_filter)
```
Comment (vmihailenco) [inline pybrake/logging.py]: Any reasons this is added here?

File: pybrake/global_notifier.py [hunk 11]
```diff
@@ -22,5 +22,6 @@ def get_global_notifier():
             return None
 
     from .notifier import Notifier
+    notifier = Notifier(**settings.AIRBRAKE)
 
-    return Notifier(**settings.AIRBRAKE)
+    return notifier
```
Comment (vmihailenco) [inline pybrake/global_notifier.py]: revert

File: pybrake/django.py (hunk 7 above)
Comment (vmihailenco) [reply pybrake/django.py]: revert

File: pybrake/django.py [hunk 12]
```diff
@@ -1,21 +1,59 @@
...
+def set_request(request):
+    setattr(threadLocal, _REQUEST_KEY, request)
+
+def get_request():
+    return getattr(threadLocal, _REQUEST_KEY, None)
+
+def request_filter(notice):
+    request = getattr(threadLocal, _REQUEST_KEY, None)
+
+    if not request:
```
Comment (vmihailenco) [inline pybrake/django.py]: `if request is None:` and remove empty line before this check

**Comment by anmic** [comment]
Fixed.

File: pybrake/django.py [hunk 13]
```diff
@@ -31,11 +59,16 @@ def template_render(self, context):
 
 class AirbrakeMiddleware:
     def __init__(self, get_response):
-        self.get_response = get_response
         self._notifier = get_global_notifier()
+        if _DJANGO:
```
Comment (vmihailenco) [inline pybrake/django.py]: remove this check - this is django

File: pybrake/django.py [hunk 14]
```diff
@@ -31,11 +59,16 @@ def template_render(self, context):
...
+            self._notifier.add_filter(request_filter)
+
         self._apm_disabled = self._notifier._apm_disabled
+        self.get_response = get_response
 
     def __call__(self, request):
+        set_request(request)
```
Comment (vmihailenco) [inline pybrake/django.py]: move after self._apm_disabled is checked

**Comment by anmic** [comment]
@vmihailenco PTAL

**Comment by vmihailenco** [comment]
Please setup https://black.readthedocs.io/en/stable/ so the code is auto-formatted
